py-polars/tests/unit/operations/test_index_of.py

```diff
@@ -0,0 +1,245 @@
+from __future__ import annotations
+
+from datetime import date, datetime, time, timedelta
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+import pytest
+from hypothesis import example, given
+from hypothesis import strategies as st
+
+import polars as pl
+
+if TYPE_CHECKING:
+    from polars._typing import IntoExpr, NonNestedLiteral
+from polars.testing import assert_frame_equal
+
+
+def assert_index_of(
+    series: pl.Series,
+    value: NonNestedLiteral | IntoExpr | None,
+    convert_to_literal: bool = False,
+) -> None:
+    """``Series.index_of()`` returns the index, or ``None`` if it can't be found."""
+    if isinstance(value, (np.number, float, int)) and np.isnan(value):
+        expected_index = None
+        for i, o in enumerate(series.to_list()):
+            if o is not None and np.isnan(o):
+                expected_index = i
+                break
+    else:
+        try:
+            expected_index = series.to_list().index(value)
+        except ValueError:
+            expected_index = None
+    if expected_index == -1:
+        expected_index = None
+    # Eager API:
+    assert series.index_of(value) == expected_index
+    # Lazy API:
+    assert pl.LazyFrame({"series": series}).select(
+        pl.col("series").index_of(value)
+    ).collect().get_column("series").to_list() == [expected_index]
+
+    # With matching dtype:
+    if not convert_to_literal:
+        return
+    value = pl.lit(value, dtype=series.dtype)
+    assert series.index_of(value) == expected_index
+    assert pl.LazyFrame({"series": series}).select(
+        pl.col("series").index_of(value)
+    ).collect().get_column("series").to_list() == [expected_index]
+
+
+@pytest.mark.parametrize("dtype", [pl.Float32, pl.Float64])
+def test_float(dtype: pl.DataType) -> None:
```

Comment:
Can we add tests to cover signed zeroes and NaNs for floats? i.e. `[0.0].index_of(-0.0)` and `[nan].index_of(-nan)` should both return a match